src/network/networkd-link.c

```diff
@@ -1511,6 +1511,15 @@ static int link_up(Link *link) {
         }
 
         if (link->network->mtu) {
+                /* IPv6 protocol requires a minimum MTU of IPV6_MTU_MIN(1280) bytes
+                   on the interface. Bump up MTU bytes to IPV6_MTU_MIN. */
+                if (socket_ipv6_is_supported() && link->network->mtu < IPV6_MIN_MTU) {
```

Comment:
Hmm, I wonder, is socket_ipv6_is_supported() actually the right choice here? Shouldn't this only be bumped if IPv6 is actrually enabled on that interface?

Or does the kernel refuse the low MTUs completely and in all cases if IPv6 is enabled in the kernel?